### PR TITLE
Implement new floor selector and my position button Web Components for Mapbox: Map Template

### PR DESCRIPTION
**Pull request by annakubackaAnku**
# What 
- Implement the new Floor-selector and My-position Web Components for the Map Template for Mapbox.

# How 
- I updated the SDK reference in the `index.html` file.
- In the `Map.jsx` file I updated the `onPositionControlCreated` function to attach the event listeners properly, according to what kind of position control is present in the Map template.
- In the `MapboxMap.jsx` file I exchanged the implementation of the standard Web SDK floor-selector and my-position button to use the new Web Components.
- A minor styling fix was needed in the `MapTemplate.scss` file for the floor-selector Web Component.

### CODE REVIEW

File: packages/map-template/src/components/Map/Map.jsx [hunk 1]
```diff
@@ -205,11 +205,20 @@ function Map({ onLocationClick, onVenueChangedOnMap }) {
      * @param {object} positionControl - MapsIndoors PositionControl instance.
      */
     const onPositionControlCreated = positionControl => {
-        positionControl.on('position_received', positionInfo => {
-            if (positionInfo.accurate === true) {
-                setUserPosition(positionInfo.position);
-            }
-        });
+        if (document.querySelector('mi-my-position')) {
```
Comment (ammapspeople) [inline packages/map-template/src/components/Map/Map.jsx:208]: Could we just as well use `positionControl.nodeName` to check for the position control type? I feel that is more clean than querying the DOM.

File: packages/map-template/src/components/Map/MapboxMap/MapboxMap.jsx [hunk 2]
```diff
@@ -45,28 +45,28 @@ function MapboxMap({ onMapView, onPositionControl }) {
     // Add Floor Selector to the Map when ready.
     useEffect(() => {
         if (mapsIndoorsInstance && mapView && !hasFloorSelector) {
-            const floorSelectorDiv = document.createElement('div');
-            new mapsindoors.FloorSelector(floorSelectorDiv, mapsIndoorsInstance);
+            const floorSelectorElement = document.createElement('mi-floor-selector');
+            floorSelectorElement.mapsindoors = mapsIndoorsInstance;
+            floorSelectorElement.primaryColor = '#FFA500';
```
Comment (ammapspeople) [inline packages/map-template/src/components/Map/MapboxMap/MapboxMap.jsx:50]: Is this the color we want to use in the Map Template?

File: packages/map-template/src/components/Map/MapboxMap/MapboxMap.jsx [hunk 3]
```diff
@@ -45,28 +45,28 @@ function MapboxMap({ onMapView, onPositionControl }) {
...
+
             mapView.getMap().addControl({
-                onAdd: () => floorSelectorDiv,
-                onRemove: () => {
-                    floorSelectorDiv.parentNode.removeChild(floorSelectorDiv);
-                }
+                onAdd: () => floorSelectorElement,
+                onRemove: function () { }
```
Comment (ammapspeople) [inline packages/map-template/src/components/Map/MapboxMap/MapboxMap.jsx:54]: Why don't we want to remove the floor selector?

**Comment by ammapspeople** [comment]
Do we want the position dot and circle on the map to be black? It used to be blue.

File: packages/map-template/src/components/Map/MapboxMap/MapboxMap.jsx (hunk 2 above)
Comment (annakubackaAnku) [reply packages/map-template/src/components/Map/MapboxMap/MapboxMap.jsx:50]: @ammapspeople ah that's a good catch, we need the color the user pass in - Where can I get that?

Comment (ammapspeople) [reply packages/map-template/src/components/Map/MapboxMap/MapboxMap.jsx:50]: Take a look at how `primaryColorState` is used in the `LocationDetails.jsx`.

**Review by ammapspeople — APPROVED** [review]
🙌

**Pull request merged**